openverse_catalog/dags/providers/provider_api_scripts/freesound.py

```diff
@@ -8,294 +8,247 @@
 
-preview_bitrates = {
-    "preview-hq-mp3": 128000,
-    "preview-lq-mp3": 64000,
-    "preview-hq-ogg": 192000,
-    "preview-lq-ogg": 80000,
-}
-
-
-def main(date="all"):
-    """This script pulls the data for a given date from the Freesound,
-    and writes it into a .TSV file to be eventually read
-    into our DB.
-
-    Required Arguments:
-
-    date:  Date String in the form YYYY-MM-DD.  This is the date for
-           which running the script will pull data.
-    """
-
-    logger.info("Begin: Freesound script")
-    licenses = ["Attribution", "Attribution Noncommercial", "Creative Commons 0"]
-    for license_name in licenses:
-        audio_count = _get_items(license_name, date)
-        logger.info(f"Audios for {license_name} pulled: {audio_count}")
-    logger.info("Terminated!")
-
-
-def _get_query_params(
-    license_name="",
-    page_number=1,
-    default_query_params=None,
-):
-    default_query_params = default_query_params or DEFAULT_QUERY_PARAMS
-    return {
-        **default_query_params,
-        "page": str(page_number),
-        "license": license_name,
+
+class FreesoundDataIngester(ProviderDataIngester):
+    batch_limit = 150
+    host = "freesound.org"
+    endpoint = f"https://{host}/apiv2/search/text"
+    providers = {"audio": prov.FREESOUND_DEFAULT_PROVIDER}
+    flaky_exceptions = (SSLError, ConnectionError)
+    preview_bitrates = {
+        "preview-hq-mp3": 128000,
+        "preview-lq-mp3": 64000,
+        "preview-hq-ogg": 192000,
+        "preview-lq-ogg": 80000,
     }
 
+    headers = {
+        "Accept": "application/json",
+    }
 
-def _get_items(license_name, date):
-    item_count = 0
-    page_number = 1
-    should_continue = True
-    default_query_params = copy.deepcopy(DEFAULT_QUERY_PARAMS)
-    try:
-        start_date = datetime.strftime(
-            datetime.fromisoformat(date), "%Y-%m-%dT%H:%M:%SZ"
-        )
-    except ValueError:
-        start_date = "*"
-    default_query_params["filter"] = f"created:[{start_date} TO NOW]"
-    while should_continue:
-        query_param = _get_query_params(
-            default_query_params=default_query_params,
-            license_name=license_name,
-            page_number=page_number,
-        )
-        batch_data = _get_batch_json(query_param=query_param)
-        if batch_data:
-            item_count = _process_item_batch(batch_data)
-            page_number += 1
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.api_key = Variable.get("API_KEY_FREESOUND")
+
+    def get_media_type(self, record: dict) -> str:
+        return constants.AUDIO
+
+    def ingest_records(self, **kwargs):
+        for license_name in [
+            "Attribution",
+            "Attribution Noncommercial",
+            "Creative Commons 0",
+        ]:
+            logger.info(f"Obtaining audio records under license '{license_name}'")
+            super().ingest_records(license_name=license_name, **kwargs)
```

Comment:
AFAIK Freesound provides only CC-licensed tracks so is this custom code needed at all? They also [mention](https://freesound.org/help/faq/#licenses) they used to include the [Sampling+ license](http://creativecommons.org/licenses/sampling+/1.0/) (now retired) and we have ingested a few of those as well but now will be omitted here.

What do you think?

Reply:
I'm glad you pointed this out! I looked more into their [API's `filter` notes](https://freesound.org/docs/api/resources_apiv2.html?highlight=license#request-parameters-text-search-parameters), and apparently we're using this entirely wrong! The `license` param is completely ignored in the request (I tried `Attribution` locally and saw that I got back a ton of different license types within the first 25 results:

```
In [14]: {x["license"] for x in r[0]}
Out[14]: 
{'http://creativecommons.org/licenses/by-nc/3.0/',
 'http://creativecommons.org/licenses/by/3.0/',
 'http://creativecommons.org/publicdomain/zero/1.0/',
 'https://creativecommons.org/licenses/by-nc/4.0/',
 'https://creativecommons.org/licenses/by/4.0/'}
```

As you mention, I don't think this logic is needed at all (and we obviously aren't even using it presently). I think it'd be best to remove it, but we could also create an issue for the stability milestone to have this filter by license type correctly. Given the nature of the script I don't think there's any advantage to doing so. What do you think @WordPress/openverse-catalog?

Reply:
It makes sense that it wasn't doing anything, especially considering we were previously using `license_name` as the param 😮 I guess the intention must have been to add it to the filter? But even so, you're right, it would be unnecessary!

I think removing this as part of this PR is completely fine, because as you point out it doesn't change the _behavior_ of the script at all 😄 Well spotted!